spec/lib/metasploit/framework/login_scanner/http_spec.rb

```diff
@@ -30,4 +35,160 @@
+        "# Response:",
+        "####################",
+        "%clr%bld%bluHTTP/1.1 302 Found\r",
+        "\r",
+        "%clr"
+      ]
+    }
+
+    let(:http_trace_colors_output) {
+      [
+        "####################",
+        "# Request:",
+        "####################",
+        "%clr%bld%bluGET / HTTP/1.1",
+        "Host: www.google.com%clr",
+        "####################",
+        "# Response:",
+        "####################",
+        "%clr%bld%grnHTTP/1.1 302 Found\r",
+        "\r",
+        "Location: https://www.google.com/?gws_rd=ssl%clr"
+      ]
+    }
+
+    let(:http_trace_single_color_output) {
+      [
+        "####################",
+        "# Request:",
+        "####################",
+        "%clr%bld%yelGET / HTTP/1.1",
+        "Host: www.google.com%clr",
+        "####################",
+        "# Response:",
+        "####################",
+        "%clrHTTP/1.1 302 Found\r",
+        "\r",
+        "Location: https://www.google.com/?gws_rd=ssl%clr"
+      ]
+    }
+
+    it 'returns a proc object when HttpTrace is set to true' do
+      expect(subject.set_http_trace_proc(true, false, nil)).to be_kind_of(Proc)
+    end
+
+    it 'should execute the proc when defined' do
+      subject.set_http_trace_proc(true, false, nil).call(sample_request, sample_response)
+      expect(@output).to eq expected_output
+    end 
+
+    it 'should give "no response received" message for nil response' do
+      subject.set_http_trace_proc(true, false, nil).call(sample_request, nil)
+      expect(@output).to eq nil_response_output
+    end
+
+    it 'should log empty message for nil request' do
+      subject.set_http_trace_proc(true, false, nil).call(nil, nil)
+      expect(@output).to eq nil_request_output
+    end
+
+    it 'should log HTTP headers only when HttpTraceHeadersOnly is set' do
+      subject.set_http_trace_proc(true, true, nil).call(sample_request, sample_response)
+      expect(@output).to eq headers_only_output
+    end
+
+    it 'should log HTTP requests and responses with body when HttpTraceHeadersOnly is unset' do
+      subject.set_http_trace_proc(true, nil, nil).call(sample_request, sample_response)
+      expect(@output).to eq expected_output
+    end
+
+    it 'should log HTTP requests and responses in the specified color' do
+      subject.set_http_trace_proc(true, false, 'blu/grn').call(sample_request, sample_response)
+      expect(@output).to eq http_trace_colors_output
+    end
+
+    it 'should only log HTTP request in color when only one colour is specified' do
+      subject.set_http_trace_proc(true, false, 'yel/').call(sample_request, sample_response)
+      expect(@output).to eq http_trace_single_color_output
+    end
+
+    it 'should only log HTTP request in color when only one color is specified without "/"' do
+      subject.set_http_trace_proc(true, false, 'yel').call(sample_request, sample_response)
+      expect(@output).to eq http_trace_single_color_output
+    end
```

Comment:
What if the option starts with a slash and then a color. So something like `/yel`?